linux/deploy_vm/ubuntu/get_mainline_kernel.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022-2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Set the base url for kernel version check"
+  ansible.builtin.set_fact:
+    mainline_kernel_url: "https://kernel.ubuntu.com/mainline"
+
+- name: "Check the kernel version exists"
+  ansible.builtin.uri:
+    url: "{{ mainline_kernel_url }}/v{{ upgrade_kernel_version }}/amd64"
+    return_content: true
+  register: check_kernel_result
+  ignore_errors: yes
+  when: upgrade_kernel_version is match('[0-9]+.[0-9a-z.-]+')
+
+- name: "Get the latest kernel version"
+  when: >
+    (upgrade_kernel_version | lower == 'latest' or
+     upgrade_kernel_version is not match('[0-9]+.[0-9a-z.-]+') or
+     check_kernel_result.status is undefined or check_kernel_result.status != 200)
```

Comment:
I wonder why not fail the task when check_kernel_result is failed. Since `upgrade_kernel_version` is from user's input, user is responsible for inputting a valid and correct kernel version. If the kernel version doesn't exists, then fail the test. Otherwise setting `upgrade_kernel_version: latest` is meaningless as user can just input an invalid kernel version it will always get the latest one. Why do we need user to set  `upgrade_kernel_version: latest` then?